src/constraints.jl

```diff
@@ -10,11 +10,21 @@ If `F`-in-`S` constraints are only not supported in specific circumstances, e.g.
 supportsconstraint(model::ModelLike, ::Type{<:AbstractFunction}, ::Type{<:AbstractSet}) = false
 
 """
-    canaddconstraint(model::ModelLike, ::Type{F}, ::Type{S})::Bool where {F<:AbstractFunction,S<:AbstractSet}
+    UnsupportedConstraint{F, S} <: CannotError
 
-Return a `Bool` indicating whether it is possible to add a constraint ``f(x) \\in \\mathcal{S}`` where ``f`` is of type `F`, and ``\\mathcal{S}`` is of type `S`.
+Constraints of type `F`-in-`S` are not supported by the model.
 """
-canaddconstraint(model::ModelLike, ::Type{<:AbstractFunction}, ::Type{<:AbstractSet}) = false
+struct UnsupportedConstraint{F<:AbstractFunction, S<:AbstractSet} <: CannotError end
```

Comment:
We should consider adding an optional text error string in case the solver has a more specific error message to return.